### DISCUSSION TITLE
How to configure logging for 1.5.0 version?

### DISCUSSION THREAD

**Discussion by Loydik** [testcontainers/testcontainers-dotnet#451] [Q&A] [answered]
Hello, I've been trying to add logging to testcontainers, I've read the docs and it says

_Logging
To enable and configure logging, set the static TestcontainersSettings.Logger property before test execution._

I was struggling to find it, but then I figured out that this functionality is not part of latest release (1.5.0), as this class was added on 24th of July.

Why are there no more releases after 1.5.0? How can we configure logging for 1.5.0?

**Comment by HofmeisterAn** [comment, marked as answer]
In version `1.5.0` logging was bound to Serilog. You'll need to provide a [appsettings.json](https://github.com/HofmeisterAn/dotnet-testcontainers/blob/master/src/DotNet.Testcontainers.Tests/appsettings.json) configuration and the necessary [sinks](https://github.com/HofmeisterAn/dotnet-testcontainers/blob/4186d347e86b246a2e30cd93910a297d6a47d7df/src/DotNet.Testcontainers.Tests/DotNet.Testcontainers.Tests.csproj#L11-L12). I recomment the latest snapshot release. It contains a lot improvements incl. `TestcontainersSettings.Logger` which supports the `ILogger`.

**Reply by Loydik** [reply]
Thank you very much!

**Reply by Loydik** [reply]
One more clarification - we assigned an ILogger instance to a static property Logger in TestcontainersSettings, however it doesn't seem like logging is working out of the box. Nothing is being logged, even though the container was started and disposed.
Are we missing some configuration? Or does it log only errors? (we have loglevel set as trace)
Thank you!

**Reply by HofmeisterAn** [reply]
> Are we missing some configuration?

It's necessary to set the [logger](https://github.com/HofmeisterAn/dotnet-testcontainers/blob/982e736b513b5d225c4263d7d0a529db2a339042/tests/DotNet.Testcontainers.Tests/Initialization.cs#L15) before you create the container instance.

> Or does it log only errors? (we have loglevel set as trace)

Log level is usually `Information`, see the [tests](https://dev.azure.com/HofmeisterAn/GitHub-Testcontainers/_build/results?buildId=2197&view=logs&j=fc799872-1c04-54a5-6e88-3d69a0bc5800&t=f07bc4a8-3cbb-5f48-bfbf-4cfe2853cb8d).

**Reply by Loydik** [reply]
@HofmeisterAn thanks, that clears things out. Maybe add that point to the Readme? It wasn't so obvious

"It's necessary to set the [logger](https://github.com/HofmeisterAn/dotnet-testcontainers/blob/982e736b513b5d225c4263d7d0a529db2a339042/tests/DotNet.Testcontainers.Tests/Initialization.cs#L15) before you create the container instance."

**Reply by ssippe** [reply]
e.g. `TestcontainersSettings.Logger = new Microsoft.Extensions.Logging.Debug.DebugLoggerProvider().CreateLogger("test");`
using [this](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Debug/) nuget package
